common/models/role.js

```diff
@@ -273,6 +273,10 @@ module.exports = function(Role) {
         process.nextTick(function() {
           callback(null, matches(modelId, userId));
         });
+      }else {
+        process.nextTick(function() {
+          callback(null, false);
+        });
```

Comment:
This will not work in the following case: one user model owns another user mode, e.g. a Seller owns the business account of Customer.